#! /usr/bin/env python

PACKAGE='foscam_8918_driver'
import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()
#       Name          Type   Level Description                       Default Min   Max
gen.add("username",   str_t, 0,    "The username for the camera.",   "change_me")
gen.add("password",   str_t, 0,    "The password for the camera.",   "change_me")
gen.add("ip_address", str_t, 0,    "The IP address of the camera..", "192.168.1.1")
gen.add("port",       str_t, 0,    "The port of the camera.",        "80")
gen.add("url_suffix", str_t, 0,    "The URL suffix for the camera.", "video.cgi?.mjpg")
gen.add("debug_show_image_window", bool_t, 0, "Show window with camera image.", False)
gen.add("reset_connection", bool_t, 0, "Connect to camera with current parameters.", False)

exit(gen.generate(PACKAGE, "foscam_8918_driver", "foscam_8918_driver"))
